enterprise_subsidy/apps/transaction/signals/handlers.py

```diff
@@ -135,14 +161,38 @@ def handle_lc_enrollment_revoked(**kwargs):
             f"Found existing Reversal: {existing_reversal} for enterprise fulfillment: "
             f"{fulfillment_uuid}. Skipping Reversal creation for Transaction: {related_transaction}."
         )
-        return
+        return False
 
     # Continue on if no reversal found
     logger.info(
         f"No existing Reversal found for enterprise fulfillment: {fulfillment_uuid}. "
-        f"Writing Reversal for Transaction: {related_transaction}."
+        f"Proceeding to attempt to write Reversal for Transaction: {related_transaction}."
     )
 
+    if not dry_run:
+        # Opportunitstically cancel any external fulfillments. [ENT-10284] Do this BEFORE checking
+        # refundability in order to prioritize complete unenrollment and system integrity even in
+        # non-refundable situations.
+        try:
+            cancel_transaction_external_fulfillment(related_transaction)
```

Comment:
Here's the actual fix for the bug.  Note this function call moved before the `unenrollment_can_be_refunded()` call below.